build.ps1

```diff
@@ -123,6 +133,12 @@ Invoke-BuildStep 'Running Restore' {
     Trace-Log ". `"$MSBuildExe`" $args"
     & $MSBuildExe @args
 
+    if ($Internal)
+    {
+        Trace-Log ". dotnet nuget disable source dotnet5-internal"
+        & dotnet nuget disable source dotnet5-internal
+    }
```

Comment:
should this be in a `finally` block? Or maybe we can run this code unconditionally _before_ https://github.com/NuGet/NuGet.Client/pull/3597/files#diff-f0e02e42e53ec40f5d3b8bed031ad09eR113, so that `^C` doesn't put is in a weird state with this?

Reply:
good question. finally block won't catch ctrl-c.
might be able to start restore as a job and then catch ctrl-c, etc...
https://social.technet.microsoft.com/Forums/windowsserver/en-US/fb409048-a607-4895-8ab3-08c2ec656c7a/trapping-ctrlc-in-powershell-v2

i don't like setting it unconditionally in the beginning to reset...because i might prefer to hand edit the nuget.config and just never check in those changes. and this would prevent that.

aagghh. not a great answer. let me noodle.

Reply:
ok. while i could go do the "job" approach and handle ctrl-c during restore...

i lean towards your second approach ... disabling the feed unconditionally in the beginning to keep you in sync.
made that change. curious what others think too... cause it isn't great.